src/python/pants/goal/run_tracker.py

```diff
@@ -398,12 +398,14 @@ def do_post(url, num_redirects_allowed):
                         cookies=cookies.get_cookie_jar(), allow_redirects=False)
       if res.status_code in {307, 308}:
         return do_post(res.headers['location'], num_redirects_allowed - 1)
-      elif res.status_code != 200:
+      elif 300 <= res.status_code < 400 or res.status_code == 401:
```

Comment:
This all might be a bit less awkward if the 1st check was for ok, ie:
```python
if res.ok:
  return True
elif res.status_code in {307, 308}:
  return do_post(res.headers['location'], num_redirects_allowed - 1)
else:
  error(f'HTTP error code: {res.status_code}. Reason: {res.reason}.')
  if 300 <= res.status_code < 400 or res.status_code == 401:
    print(f'Use `path/to/pants login --to={auth_provider}` to authenticate '
          'against the stats upload service.', file=sys.stderr)
  return False

Reply:
@jsirois this won't work, since 307 and 307 are also considered ok responses, and we want to treat those as errors.
https://github.com/psf/requests/blob/master/requests/models.py#L702
https://github.com/psf/requests/blob/a4c18cd733f97b5659a29589432d8a39e7a0de87/requests/models.py#L933-L936

Reply:
There is actually a test to validate that HTTP 307 results in an error:
https://github.com/pantsbuild/pants/pull/8102/files#diff-9e6ae65d103fa02ecc6e5f0ce5eeb3d7L57